src/lightning/pytorch/loops/fit_loop.py

```diff
@@ -226,17 +352,14 @@ def on_run_start(self) -> None:
     def on_advance_start(self) -> None:
         """Prepares the dataloader for training and calls the hook ``on_train_epoch_start``"""
         trainer = self.trainer
-        model = trainer.lightning_module
 
-        # reset train dataloader
-        if not self._is_fresh_start_epoch and trainer._data_connector._should_reload_train_dl:
-            log.debug(f"{self.__class__.__name__}: resetting train dataloader")
-            trainer.reset_train_dataloader(model)
-        self._is_fresh_start_epoch = False
+        # might need to setup data again depending on `trainer.reload_dataloaders_every_n_epochs`
+        self.setup_data()
```

Comment:
This means we need to exit early if not.
I see the check inside but does `not self._is_fresh_start_epoch` correspond to ` self._combined_loader is not None`?